### ISSUE TITLE
No error generated when reactor file does not exist.

### ISSUE THREAD

**Issue by scubahub** [saltstack/salt#36579]
### Description of Issue/Question
Ref https://github.com/saltstack/salt/issues/36114, since I am still having issues with some scheduled jobs returning data that the reactor sls file files to render (json->yaml->json issue) and I realized you can write reactor files in pure python, I decided to redo my returner/reactor/runner implementation shown in that issue so the reactor is python.

When trying to test the new reactor, the name of the reactor changed from scheduled-job-return-reactor.sls to scheduled-job-return-reactor.py (change in file extension) and I removed the old file (I since realized I did not need to change the file extension, old habits).  However I forgot to update the master config to direct the event to the new reactor file so it still listed the old file name which no longer exists.

Turns out no errors are generated when the reactor config references a file that does not exist.  Instead all that I see is

```
2016-09-26 18:54:10,829 [salt.utils.event ][DEBUG   ][24493] Sending event - data = {'_stamp': '2016-09-26T18:54:10.829168', ...}
2016-09-26 18:54:10,831 [salt.utils.reactor][DEBUG   ][24493] Gathering reactors for tag scheduled-job-return
2016-09-26 18:54:10,831 [salt.utils.reactor][DEBUG   ][24493] Compiling reactions for tag scheduled-job-return
```

but the reactor never runs.  I eventually figured out that interspersed with lots of unrelated stuff in the log there is a sequence of these messages from the same thread

```
2016-09-26 19:08:27,988 [salt.utils.event ][TRACE   ][24493] _get_event() waited 5000 seconds and received nothing
2016-09-26 19:08:32,992 [salt.utils.event ][TRACE   ][24493] _get_event() waited 5000 seconds and received nothing
2016-09-26 19:08:37,993 [salt.utils.event ][TRACE   ][24493] _get_event() waited 5000 seconds and received nothing
2016-09-26 19:08:42,996 [salt.utils.event ][TRACE   ][24493] _get_event() waited 5000 seconds and received nothing
2016-09-26 19:08:47,998 [salt.utils.event ][TRACE   ][24493] _get_event() waited 5000 seconds and received nothing
2016-09-26 19:08:53,003 [salt.utils.event ][TRACE   ][24493] _get_event() waited 5000 seconds and received nothing
```

That made me reexamine everything more closely and I finally realized the issue was the reactor file name was wrong in the reactor config for the master.  This condition should produce an error.

### Versions Report
```
Salt Version:
           Salt: 2015.8.11

Dependency Versions:
         Jinja2: 2.7.2
       M2Crypto: 0.21.1
           Mako: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.7.0
         Python: 2.7.5 (default, Nov 20 2015, 02:00:19)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 1.5
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
        libgit2: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.7
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: Not Installed
        timelib: Not Installed

System Versions:
           dist: centos 7.2.1511 Core
        machine: x86_64
        release: 3.10.0-229.14.1.el7.x86_64
         system: CentOS Linux 7.2.1511 Core
```

**Comment by Ch3LL** [comment]
@scubahub I am able to replicate this on 2016.3.3 as well and all I see is:

```
2016-09-27 17:12:37,203 [salt.utils.reactor                       ][DEBUG   ][771] Gathering reactors for tag salt/minion/7d99677b3bf8/start
2016-09-27 17:12:37,203 [salt.utils.reactor                       ][DEBUG   ][771] Compiling reactions for tag salt/minion/7d99677b3bf8/start

2016-09-27 17:13:22,769 [salt.config                              ][DEBUG   ][750] Reading configuration from /etc/salt/master
2016-09-27 17:13:22,828 [salt.config                              ][DEBUG   ][750] Including configuration from '/etc/salt/minion.d/_schedule.conf'
```

When it does not exist. Looks like we need add an error message when the file does not exist. Thanks! 

Here is a docker container to replicate the issue:

1. `docker run -it -v /home/ch3ll/git/salt:/testing/ ch3ll/issues:36579 /bin/bash` (where /home/ch3ll/git/salt/ is a local git clone of salt)
2. `salt-master -d`
3. `salt-minion -d`
4. `tail -f /var/log/salt/master`

And you will see the logs hang for a sec where its attempting to gather reactors for salt/minion/*/start

**Comment by cachedout** [comment]
This has always bothered me too. Fixed in #36797
